graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -61,6 +65,45 @@ def _validate_macro_ast_with_macro_directives(schema, ast, macro_directives):
                 .format(directive_definition.name, len(macro_data)))
 
 
+def _validate_macro_ast_directives(ast, inside_fold_scope=False):
+    """Check that the macro is using non-macro direcives properly."""
+    subselection_inside_fold_scope = inside_fold_scope
+    directives_with_no_restrictions = {
+        FilterDirective.name,
+        TagDirective.name,
+        OptionalDirective.name,
+        RecurseDirective.name,
+        MacroEdgeDefinitionDirective.name,
+    }
+    for directive in ast.directives:
+        name = directive.name.value
+        if name in directives_with_no_restrictions:
+            pass
```

Comment:
Are you sure you want `pass` here vs. an explicit `continue`?

Reply:
Yes. The `if/elif/else` block explicitly covers all cases. A `continue` is equivalent with the `pass` but makes the control flow more complicated. If there's any code added at the end of the for loop, there's no reason it shouldn't apply to the `directives_with_no_restrictions`.